java/vector/src/main/java/org/apache/arrow/vector/BaseNullableFixedWidthVector.java

```diff
@@ -241,6 +241,10 @@ public boolean allocateNewSafe() {
    public void allocateNew(int valueCount) {
       long valueBufferSize = valueCount * typeWidth;
       long validityBufferSize = getSizeFromCount(valueCount);
+      if (typeWidth == 0) {
```

Comment:
Note to myself:
This special if case for bit vector is bit of an eye sore. Look into this.